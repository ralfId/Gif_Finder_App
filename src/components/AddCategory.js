import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategory = ({setCategory}) => {
  const [inputValue, setInputValue] = useState('');

  const handleOnChangeInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) =>{
      e.preventDefault();
      console.log('handleSubmit', inputValue);
      if ( inputValue.trim().length > 2 ) {
        setCategory(catg => [inputValue, ...catg] );
      }
      setInputValue('');
  };

  return (
    <form onSubmit={handleSubmit}>
        <p>{inputValue}</p>
        <input onChange={handleOnChangeInput} value={ inputValue } type="text" />

    </form>
  );
};

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
