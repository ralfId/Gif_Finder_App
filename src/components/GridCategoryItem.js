import React from 'react'
import PropTypes from 'prop-types';

export const GridCategoryItem = ({ title, img }) => {
    return (
        <div className='card animate__fadeInLeft'>
            <img src={img} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GridCategoryItem.protoTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired
};