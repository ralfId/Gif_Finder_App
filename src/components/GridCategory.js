import React, { useState, useEffect } from 'react'
import { GetGif } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GridCategoryItem } from './GridCategoryItem';

export const GridCategory = ({ category }) => {


    // const [gifImg, setGifImg] = useState([])
    const {data:gifImg, loading} = useFetchGifs(category);
    
    // useEffect(() => {
    //     GetGif(category).then(setGifImg);
    // }, [category]);

    

    return (
        <>
            <h3>{category}</h3>
            {loading && <p>Loading...</p>}
            <div className='card-container animate__zoomIn'>
                {
                    gifImg.map(img => (
                        <GridCategoryItem
                            key={img.id} {...img} />
                    ))
                }
            </div>


        </>
    )
}
