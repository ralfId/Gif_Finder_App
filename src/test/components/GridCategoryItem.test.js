import React from 'react'
import { shallow } from "enzyme";
import { GridCategoryItem } from "../../components/GridCategoryItem";

describe('Test for GridCategoryItem.js', () => {
    
    const title = 'Lorem ipsum';
    const img = 'https://www.imagesgifs.com'

    const wrapper = shallow( <GridCategoryItem img={img}  title={title}/> )

    test('should show a correct snapshop for GridCategoryItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim()).toBe(title);
    })

    test('debe mostrar la url de la imagen', () => {
        const image = wrapper.find('img');
        expect(image.prop('src')).toBe(img);
    })
    
    test('El contenedor <div></div> debe tene la clase css "animate__fadeInLeft"', () => {
        const div = wrapper.find('div');
        const hasAnimatedClass = div.prop('className').includes('animate__fadeInLeft');
        expect(hasAnimatedClass).toBe(true);
    })
    
      
       

    
});