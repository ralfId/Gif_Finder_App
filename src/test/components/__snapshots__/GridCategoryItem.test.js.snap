// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GridCategoryItem.js should show a correct snapshop for GridCategoryItem component 1`] = `
<div
  className="card animate__fadeInLeft"
>
  <img
    alt="Lorem ipsum"
    src="https://www.imagesgifs.com"
  />
  <p>
    Lorem ipsum
  </p>
</div>
`;
